packages/flutter/lib/src/widgets/form.dart

```diff
@@ -56,8 +65,23 @@ class FormState extends State<Form> {
   int _generation = 0;
   Set<FormFieldState<dynamic>> _fields = new Set<FormFieldState<dynamic>>();
 
-  /// Called when a form field has changed. This will cause all form fields
-  /// to rebuild, useful if form fields have interdependencies.
+  @override
+  void dependenciesChanged() {
+    super.dependenciesChanged();
+    final ModalRoute<Null> route = ModalRoute.of(context);
+    if (route != null && route.isCurrent)
+      route.scopedWillPopCallback = config.onWillPop;
```

Comment:
What if there are two forms in a route?  Also, don't we need to unregister the callback on dispose?